script.js

```diff
@@ -15,3 +15,187 @@ closeBtn.addEventListener('click', toggle);
+  popupImage.src = project.featuredImage;
+
+  // Add technologies to the list
+  popupLangs.innerHTML = '';
+  project.technologies.forEach((technology) => {
+    const li = document.createElement('li');
+    li.textContent = technology;
+    popupLangs.appendChild(li);
+  });
+
+  // Set the link URLs
+  liveLink.href = project.liveLink;
+  sourceLink.href = project.sourceLink;
+
+  // Show the popup
+  popupContainer.style.display = 'block';
+}
+
+// Function to open the popup for desktop version and populate project details
+function openPopups(projects) {
+  const popupContainer = document.querySelector('.popup-containers');
+  const popupHeading = popupContainer.querySelector('.pop-headers');
+  const popupImage = popupContainer.querySelector('.pop-imgs img');
+  const popupText = popupContainer.querySelector('.pop-texts p');
+  const popupText2 = popupContainer.querySelector('.text-2');
+  const popupLangs = popupContainer.querySelector('.pop-langss-new ul');
+  const popupLangs2 = popupContainer.querySelector('.pop-langss ul');
+  const liveLink = popupContainer.querySelector('.project-links');
+  const sourceLink = popupContainer.querySelector('.github-links a');
+
+  // Populate project details
+  popupHeading.textContent = projects.name;
+  popupText.textContent = projects.description;
+  popupText2.textContent = projects.description;
+  popupImage.src = projects.featuredImage;
+
+  // Add technologies to the list
+  popupLangs2.innerHTML = '';
+  projects.technologies2.forEach((technology2) => {
+    const lis = document.createElement('li');
+    lis.textContent = technology2;
+    popupLangs2.appendChild(lis);
+  });
+
+  popupLangs.innerHTML = '';
+  projects.technologies.forEach((technology) => {
+    const li = document.createElement('li');
+    li.textContent = technology;
+    popupLangs.appendChild(li);
+  });
+
+  // Set the link URLs for mobile version
+  liveLink.href = projects.liveLink;
+  sourceLink.href = projects.sourceLink;
+
+  // Show the popup for mobile version
+  popupContainer.style.display = 'block';
+}
+
+// Function to close the popup for mobile version
+function closePopup() {
+  const popupContainer = document.querySelector('.popup-container');
+  popupContainer.style.display = 'none';
+}
+
+// Function to close the popup for desktop version
+function closePopups() {
+  const popupContainer = document.querySelector('.popup-containers');
+  popupContainer.style.display = 'none';
+}
+
+// Attach click event listeners to buttons
+const mobileButtons = document.querySelectorAll('.button');
+const desktopButtons = document.querySelectorAll('.buttons');
+
+mobileButtons.forEach((button, index) => {
+  button.addEventListener('click', () => {
+    openPopup(project[index]); // Open the popup for the corresponding project
+  });
+});
+
+desktopButtons.forEach((button, index) => {
+  button.addEventListener('click', () => {
+    openPopups(projects[index]); // Open the popup for the corresponding project
+  });
+});
```

Comment:
- [ ] There seems to be a bit of a bug here as the popup is being displayed when the page loads and this causes a bad UX so please make sure to fix this.
- Here's how it currently behaves:
  ![firefox_4XvhHLn1lE](https://github.com/Omwomo/Microverse-web-portfolio-project/assets/104968236/47a4d748-17f7-460d-aa09-bb4da2933a3f)